darts/ad/anomaly_model.py

```diff
@@ -0,0 +1,362 @@
+            return anomaly_scores
+
+
+class FilteringAnomalyModel(AnomalyModel):
+    def __init__(self, filter: FilteringModel, scorer: Union[Scorer, Sequence[Scorer]]):
+        """Filtering Anomaly Model
+
+        Parameters
+        ----------
+        model : Filtering
+            A filtering model from Darts that will be used to filter the actual time series
+        scorer : Scorer
+            A scorer that will be used to convert the actual and filtered time series to
+            an anomaly score time series
+        """
+
+        super().__init__()
+
+        raise_if_not(
+            isinstance(filter, FilteringModel),
+            f"Model must be a darts.models.filtering not a {type(filter)}",
+        )
+        self.filter = filter
+
+        if isinstance(scorer, Sequence):
+            self.scorers = scorer
+        else:
+            self.scorers = [scorer]
+
+        for scorer in self.scorers:
+            raise_if_not(
+                isinstance(scorer, Scorer),
+                f"Scorer must be a Darts Scorer not a {type(scorer)}",
+            )
+
+    def fit(
+        self, series: TimeSeries, filter_fit_params: Optional[Dict[str, Any]] = None
+    ):
+        """Train the filter and the scorer(s) on the given time series.
+
+        Parameters
+        ----------
+        series : Darts TimeSeries
+        filter_fit_params: dict, optional
+            parameters of the Darts `.fit()` filtering model
+
+        Returns
+        -------
+        self
+            Fitted Anomaly model (filtering model and scorer(s))
+        """
+
+        if filter_fit_params is None:
+            filter_fit_params = {}
+
+        # fit filtering model
+        if hasattr(self.filter, "fit"):
+            # TODO: check if filter is already fitted (for now fit it regardless -> only Kallman)
+            self.filter.fit(series, **filter_fit_params)
+
+        # fit scorer model
+        for scorer in self.scorers:
+            if hasattr(scorer, "fit"):
+                pred = self.filter.filter(series)
+                scorer.fit(pred, series)
+
+    def score(self, series: TimeSeries, filter_params: Optional[Dict[str, Any]] = None):
+        """Filters the given input time series with the filtering model, and applies the scorer(s)
+        on the filtered time series and the given input time series. Outputs the anomaly score of
+        the given input time series.
+
+        Parameters
+        ----------
+        series : Darts TimeSeries
+        filter_params: dict, optional
+            parameters of the Darts `.filter()` filtering model
+
+        Returns
+        -------
+        Darts TimeSeries
+            Anomaly score time series
+        """
+
+        if filter_params is None:
+            filter_params = {}
+
+        pred = self.filter.filter(series, **filter_params)
+
+        anomaly_scores = []
+
+        for i, scorer in enumerate(self.scorers):
+            anomaly_scores.append(scorer.compute(pred, series))
```

Comment:
What's the best here for downstream use (e.g. with detectors): return a list of univariate series, or a single multivariate series?

Reply:
The problem is that scores do not return the same length because of windowing (for example, WassersteinScorer will return a shorter anomaly score than L2Scorer). So we can not return a multivariate time series containing the anomaly scores of the scorers (unless we add Nan values to make them the same length?)